### ISSUE TITLE
Check for resources

### ISSUE THREAD

**Issue by bbruun** [weaveworks/eksctl#454]
During `eksctl create cluster ...` it will fail to create the cluster and a reverse `eksctl delete cluster ...` fails to remove all dependencies already created before `create` failed due to limitations on NAT gateways, VPC's etc. etc.

Could a pre-create check be implemented eg in the `eksctl create cluster ...` procedure or get a `--dry-run` option that will check what will be created and if it will faile due to current limitations on the IAM account used?

Also the `eksctl delete cluster ...` as see in issue #105 is a problem when the create procedure fails.

**Comment by errordeveloper** [comment]
I am not entirely clear on what is the issue. Could you please illustrate this a little more? Error messages from the complete session (try to create, fail, try to delete), and/or screenshots of CloudFormation console, would be very helpful.
Also, what is the limitation of NAT gateway that you are referring to?

**Comment by bbruun** [comment]
I tried the `eksctl create cluster...` and hit a limit on NatGateway (I think it was). I then deleted the cluster with `eksctl delete cluster <name>` and waited. 
After a while it was done and didn't show up in AWS EKS.

`eksctl delete cluster ...` didn't delete the NatGateway that it failed to created due to the limit so I had to delete it manually + another one from a test VPC that wasn't in use.

Then I tried creating the cluster again. This time I was hit by 2 other limits and I had to do a `eksctl delete cluster ...` again... and cleanup further.

It worked in a new region where we only had 1 other cluster created with eksctl, but this leaves the 2 items I'm pointing out, one being a bug IMHO:
1. Why does `eksctl delete cluster ...` not delete all the items it created on failure?
2. Could `eksctl` check for limits before it starts creating the cluster. It knows what is to be created and if one of the things it needs is maxed out on limits then it should just warn the user and stop aka not start creating the cluster and then failed to delete it as mentiond in the item above?

I hope this clarifies what I asked about.
